### ISSUE TITLE
CategoryNodes sorting should ignore spaces

### ISSUE THREAD

**Issue by jim** [rubyforgood/circulate#1044]
**Describe the bug**
`CategoryNode` is sorted by `sort_name`, which is a space-delimited list of all of the categories in that category's path. So a Category called "Saws" that is under "Power Tools" would have the `sort_name` of "power tools saws".

This is an issue because Postgres ignored spaces when sorting UTF-8 values ([it's complicated](https://stackoverflow.com/questions/34537982/incorrect-sort-collation-order-with-spaces-in-postgresql-9-4)), so we have a situation where with these categories:

```
Browse Seeds  sort_name: "browse seeds"
    Vegetable     sort_name: "browse seeds vegetable"
        Tomatoes  sort_name: "browse seeds vegetable tomatoes"
        Squash      sort_name: "browse seeds vegetable squash"
    Vegetables    sort_name: "browse seeds vegetables"
```

we end up with this order from the database:

```
Browse Seeds  sort_name: "browse seeds"
    Vegetable     sort_name: "browse seeds vegetable"
    Vegetables    sort_name: "browse seeds vegetables"
        Squash      sort_name: "browse seeds vegetable squash"
        Tomatoes  sort_name: "browse seeds vegetable tomatoes"
```

**Expected behavior**

Ideally, we want spaces to be observed when this sort occurs. The link above to Stack Overflow has some possible ways to address this.

**Comment by jpeace-pro** [comment]
What is the desired sort order? Would it be:
```
Browse Seeds  sort_name: "browse seeds"
    Vegetable     sort_name: "browse seeds vegetable"
    Squash      sort_name: "browse seeds vegetable squash"
    Tomatoes  sort_name: "browse seeds vegetable tomatoes"
    Vegetables    sort_name: "browse seeds vegetables"
```
?

**Comment by phinze** [comment]
Yep that's the right order, but Squash and Tomatoes show up indented under Vegetable.

You can mess around with the category hierarchy on the settings page to test things out: `Librarian View -> Settings -> Categories`

**Comment by jpeace-pro** [comment]
Is this still an issue? I created categories for seeds, vegetable, squash, tomatoes, vegetables, but it is showing in the "correct" order:
<img width="434" alt="image" src="https://github.com/user-attachments/assets/afe94bb2-69a6-4f06-960e-0a8aa458ce3b">

I tested in `models/category_node.rb` with this code:
```
scope :sorted, -> { order(Arel.sql("replace(sort_name, ' ', '~') ASC")) }
```

which produced the wrong sort order:
<img width="434" alt="image" src="https://github.com/user-attachments/assets/9bf0c6b4-0f42-491f-8111-eaa9095f63cd">

**Comment by jim** [comment]
I checked and while I can't get it to reproduce locally, I can do it in production still (Tomato and Squash are children of Vegetable):

<img width="209" alt="Screenshot 2024-11-03 at 10 42 13 PM" src="https://github.com/user-attachments/assets/eefda201-16a6-402d-9963-809f09ac7fb8">

I checked and the coalition in production is `en_US.UTF-8`, but locally for me it's `C`. So I think it we change the local database to that collation we should hopefully be able to reproduce the issue.

In terms of how to actually fix the issue, a migration to update the collation for this column sounds good. Since `category_nodes` is a view, it might mean that the view definition itself might need to be changed.
